Translate introducing jsx (#47)
reactjs/pt-BR.reactjs.org: content/docs/introducing-jsx.md

---
id: introducing-jsx
title: Introduzindo JSX
permalink: docs/introducing-jsx.html
prev: hello-world.html
next: rendering-elements.html
---

Considere esta declaração de variável:

```js
const element = <h1>Hello, world!</h1>;
```

Esta sintaxe estranha de tags não é uma string, nem HTML.

> É chamada JSX, e é uma extensão de sintaxe para JavaScript. Recomendamos usá-la com o React para descrever como a UI deveria parecer. JSX pode lembrar uma linguagem de template, mas vem com todo o poder do JavaScript.

Reviewer: ```suggestion
É chamada JSX e é uma extensão de sintaxe para JavaScript. Recomendamos usar JSX com o React para descrever como a UI deveria parecer. JSX pode lembrar uma linguagem de template, mas que vem com todo o poder do JavaScript.
```

JSX produz "elementos" do React. Nós iremos explorar a renderização para o DOM na [próxima seção](/docs/rendering-elements.html). Abaixo você descobrirá o básico de JSX necessário para começar.

### Por que JSX? {#why-jsx}

> O React adota o fato de que a lógica de renderização é inerentemente acoplada com outras lógicas de UI: como eventos são manipulados, como o state muda com o tempo, e como os dados são preparados para exibição.

Reviewer: ```suggestion
O React adota o fato de que a lógica de renderização é inerentemente acoplada com outras lógicas de UI: como eventos são manipulados, como o state muda com o tempo e como os dados são preparados para exibição.
```

> O React abraça o fato de que a lógica de renderização é inerentemente acoplada com outras lógicas de UI: como eventos são manipulados, como o state muda com o tempo, e como os dados são preparados para exibição.

Reviewer 2: Maybe we should translate _embraces_ as _adota_ since I believe this is the context in the phrase, what do you think?

Reviewer 2: Maybe we should translate _embraces_ as _adota_ since I believe this is the context in the phrase, what do you think?

Author: Indeed, _adota_ is a better translation for _embraces_. :)

> Ao invés de separar *tecnologias* artificialmente colocando markup e lógica em arquivos separados, o React [separa *conceitos*](https://pt.wikipedia.org/wiki/Separa%C3%A7%C3%A3o_de_conceitos) com unidades pouco acopladas chamadas "componentes" que contém ambos. Voltaremos aos componentes em 

Reviewer: ```suggestion
Ao invés de separar *tecnologias* artificialmente colocando markup e lógica em arquivos separados, o React [separa *conceitos*](https://pt.wikipedia.org/wiki/Separa%C3%A7%C3%A3o_de_conceitos) com unidades pouco acopladas chamadas "componentes" que contém em ambos. Voltaremos aos componentes em 
```

Author: @glaucia86 Acredito que "contain both"  aqui tem o sentido de "conter markup e lógica", certo? Acho que "em ambos" não encaixa, já que são os componentes que contém os dois. 

Talvez alterar para "contém tanto o markup quanto a lógica"?

> Ao invés de separar *tecnologias* artificialmente colocando markup e lógica em arquivos separados, o React [separa *conceitos*](https://pt.wikipedia.org/wiki/Separa%C3%A7%C3%A3o_de_conceitos) com unidades livremente acopladas chamadas "componentes" que contém ambos. Voltaremos aos componentes em 

Reviewer 2: _"loosely coupled"_ is a hard one to translate, but I believe the meaning of _"loosely coupled units"_ here it's a synonym to **weak** coupled. Maybe a better way to translate it would be _"unidades pouco acopladas"_.

Author: I thought of "_unidades fracamente acopladas_", but "_unidades pouco acopladas_" is just as good.

>  [outra seção](/docs/components-and-props.html), mas se você ainda não está confortável em usar markup em JS, [esta talk](https://www.youtube.com/watch?v=x7cQ3mrcKaY) pode convencer você do contrário.

Reviewer: ```suggestion
[outra seção](/docs/components-and-props.html). Mas, se você ainda não está confortável em usar markup em JS, [esta palestra](https://www.youtube.com/watch?v=x7cQ3mrcKaY) pode convencer você do contrário.
```

> O React [não requer](/docs/react-without-jsx.html) o uso do JSX, mas a maioria das pessoas acha prático como uma ajuda visual quando se está trabalhando com uma UI dentro do código em JavaScript. Ele permite ao React mostrar mensagens mais úteis de erro e aviso.

Reviewer: ```suggestion
O React [não requer](/docs/react-without-jsx.html) o uso do JSX. Porém, a maioria das pessoas acha prático como uma ajuda visual quando se está trabalhando com uma UI dentro do código em JavaScript. Ele permite ao React mostrar mensagens mais úteis de erro e aviso.
```

);
```

> Você pode por qualquer [expressão JavaScript](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Guide/Expressions_and_Operators#Expressions) válida dentro das chaves em JSX. Por exemplo, `2 + 2`, `user.firstName`, ou `formatName(user)` são todas expressões JavaScript válidas.

Reviewer 2: ```suggestion
Você pode utilizar qualquer [expressão JavaScript](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Guide/Expressions_and_Operators#Expressions) válida dentro das chaves em JSX. Por exemplo, `2 + 2`, `user.firstName`, ou `formatName(user)` são todas expressões JavaScript válidas.
```

Author: Maybe "_inserir_"?

Reviewer 2: "Inserir" is better indeed!

[](codepen://introducing-jsx)

> Separamos o JSX em múltiplas linhas para melhorar a legibilidade. Mesmo que não seja requerido, quando fizer isso, também recomendamos cercá-lo com parênteses para evitar as armadilhas da [inserção automática de ponto-e-vírgula](http://stackoverflow.com/q/2846283).

Reviewer 2: ```suggestion
Separamos o JSX em múltiplas linhas para melhorar a legibilidade. Mesmo que não seja obrigatório, quando fizer isso, também recomendamos colocar dentro de parênteses para evitar as armadilhas da [inserção automática de ponto-e-vírgula](http://stackoverflow.com/q/2846283).
```

Reviewer 2: Just opinion about _"cercá-lo"_, please tell me if you think it's best this way 😄
I believe with time we will get a better sense about what to use to translate those difficult terms.

Author: Yeah, that was a tough one. I tried "_envolver_" and "_circular_", but none of those seemed right. "_colocar dentro de_" makes sense, though.

### JSX Também é uma Expressão {#jsx-is-an-expression-too}

> Depois da compilação, as expressões em JSX se transformam em chamadas normais de funções em JavaScript e são avaliadas para objetos JavaScript.

Reviewer 2: ```suggestion
Depois da compilação, as expressões em JSX se transformam em chamadas normais de funções que retornam objetos JavaScript.
```

Reviewer 2: Here the term _"evaluate"_ means more as a return of a function other than _"avaliar"_.

Author: True!

}
```

### Especificando Atributos com JSX {#specifying-attributes-with-jsx}

Você pode usar aspas para especificar strings literais como atributos:

```js
const element = <div tabIndex="0"></div>;
```

Você também pode usar chaves para incorporar uma expressão JavaScript em um atributo:

```js
const element = <img src={user.avatarUrl}></img>;
```

Não envolva chaves com aspas quando estiver incorporando uma expressão JavaScript em um atributo. Você deveria ou usar aspas (para valores em string) ou chaves (para expressões), mas não ambos no mesmo atributo.

>**Atenção:**
>
>Como JSX é mais próximo de JavaScript que do HTML, o React DOM usa `camelCase` como convenção para nomes de propriedades ao invés dos nomes de atributos do HTML.
>
>Por exemplo, `class` se transforma em [`className`](https://developer.mozilla.org/pt-BR/docs/Web/API/Element/className) em JSX, e `tabindex` se transforma em [`tabIndex`](https://developer.mozilla.org/pt-BR/docs/Web/API/HTMLElement/tabIndex).

> ### Especificando Filhos com JSX {#specifying-children-with-jsx}

Reviewer 2: I believe we will need to discuss in the glossary how to translate _"children"_. Like _"state"_ and _"props"_, it's a main concept of React. Maybe something like _"elementos filhos"_. Just _"Filhos"_ seems odd to me.

Author: "_children_" translated as "_filhos_" seems odd to me also, but it is the term used in Computer Science for [data structure](https://pt.wikipedia.org/wiki/%C3%81rvore_(estrutura_de_dados)).

Change it to "_elementos filhos_" seems more accurate to its meaning.

};
```

Estes objetos são chamados "Elementos React". Você pode imaginá-los como descrições do que você quer ver na tela. O React lê esses objetos e os usa para construir o DOM e deixá-lo atualizado.

Exploraremos a renderização de elementos React no DOM na próxima seção.

>**Dica:**
>

> >Recomendamos o uso da [definição de linguagem "Babel"](http://babeljs.io/docs/editors) no seu editor preferido para que ambos os códigos em ES6 e JSX seja devidamente realçados. Este website usa o esquema de cores [Oceanic Next](https://labs.voronianski.com/oceanic-next-color-scheme/) no qual é compatível o mesmo.

Reviewer 2: ```suggestion
>Recomendamos o uso da [definição de linguagem "Babel"](http://babeljs.io/docs/editors) no seu editor preferido para que ambos os códigos em ES6 e JSX sejam devidamente realçados. Este website usa o esquema de cores [Oceanic Next](https://labs.voronianski.com/oceanic-next-color-scheme/) no qual é compatível o mesmo.
```

> >Recomendamos o uso da [definição de linguagem "Babel"](http://babeljs.io/docs/editors) no seu editor preferido para que ambos os códigos em ES6 e JSX sejam devidamente realçados. Este website usa o esquema de cores [Oceanic Next](https://labs.voronianski.com/oceanic-next-color-scheme/) no qual é compatível o mesmo.

Reviewer: ```suggestion
Recomendamos o uso da [definição de linguagem "Babel"](http://babeljs.io/docs/editors) no seu editor preferido para que ambos os códigos em ES6 e JSX sejam devidamente realçados. Este website usa o esquema de cores [Oceanic Next](https://labs.voronianski.com/oceanic-next-color-scheme/) no qual é compatível com o mesmo.
```

Author: @glaucia86 Se tirarmos `>`, deixa de ser uma citação, não?
